### PR TITLE
Include unrecognized components

### PR DESCRIPTION
**Pull request by jfschwarz**
Implements #665: Components that react-docgen does not recognize as react-components will still be included in the style guide. The `displayName` for such components is guessed based on the file name.

I also added an example of an unrecognizable component to the basic example.

### CODE REVIEW

File: loaders/utils/getProps.js [hunk 1]
```diff
@@ -102,5 +102,11 @@ module.exports = function getProps(doc, filepath) {
 		});
 	}
 
+	if (!doc.displayName && filepath) {
+		// The file name is our best guess for the exported component's display name
```
Comment (sapegin) [inline loaders/utils/getProps.js]: I think we should use the folder name if the file name is `index.*`.

File: examples/basic/src/components/WrappedCounterButton/Readme.md [hunk 2]
```diff
@@ -0,0 +1,6 @@
+react-docgen is not able to parse prop types of enhanced components that are imported from other files.
```
Comment (sapegin) [inline examples/basic/src/components/WrappedCounterButton/Readme.md]: I’d move this component to the `sections` example, it’s too “low-level” for the basic example.

File: examples/basic/src/components/WrappedCounterButton/Readme.md [hunk 3]
```diff
@@ -0,0 +1,6 @@
...
+We'll add style guide sections for such components nevertheless.
```
Comment (sapegin) [inline examples/basic/src/components/WrappedCounterButton/Readme.md]: “Styleguidist will add them to your style guide anyway” or something like that. They are not *sections*.

File: loaders/utils/getProps.js (hunk 1 above)
Comment (jfschwarz) [reply loaders/utils/getProps.js]: good idea

**Comment by jfschwarz** [comment]
It will still show the warning.

File: loaders/utils/getProps.js (hunk 1 above)
Comment (sapegin) [reply loaders/utils/getProps.js]: Or even better:

https://github.com/nerdlabs/react-docgen-displayname-handler/blob/8cd141064910eb429f18e03e631a25cec371330a/source/index.js#L51-L66

**Comment by sapegin** [comment]
> It will still show the warning.

That’s good I think. Probably we should mention the `ignore` option there too. But it may be another PR ;-)

**Comment by jfschwarz** [comment]
@sapegin I've implemented the changes you requested.

File: loaders/utils/getNameFromFilePath.js [hunk 4]
```diff
@@ -0,0 +1,16 @@
...
+		const parts = path.dirname(filePath).split(path.sep);
+		displayName = parts[parts.length - 1];
+	}
+
+	return displayName
+		.charAt(0)
+		.toUpperCase()
+		.concat(displayName.slice(1))
+		.replace(/-([a-z])/, (_, match) => match.toUpperCase());
+};
```
Comment (sapegin) [inline loaders/utils/getNameFromFilePath.js]: `_.upperFirst(_.camelCase(displayName))` would do the same but more readable.

File: loaders/utils/getNameFromFilePath.js [hunk 5]
```diff
@@ -0,0 +1,16 @@
+'use strict';
+const path = require('path');
+
+module.exports = function getNameFromFilePath(filePath) {
+	let displayName = path.basename(filePath, path.extname(filePath));
+	if (displayName === 'index') {
+		const parts = path.dirname(filePath).split(path.sep);
+		displayName = parts[parts.length - 1];
```
Comment (sapegin) [inline loaders/utils/getNameFromFilePath.js]: Isn’t it the same as `path.basename(path.dirname(filePath))`?

**Comment by jfschwarz** [comment]
@sapegin Sure, no problem.

**Comment by sapegin** [comment]
Looks good now, merged! 🍕

**Pull request merged**